kroxylicious-kms-provider-hashicorp-vault/src/test/java/io/kroxylicious/kms/provider/hashicorp/vault/config/InsecureTlsTest.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kms.provider.hashicorp.vault.config;
+
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.Optional;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+
+import org.junit.jupiter.api.Test;
+
+import io.kroxylicious.proxy.config.tls.InsecureTls;
+import io.kroxylicious.proxy.config.tls.Tls;
+
+import static io.kroxylicious.kms.provider.hashicorp.vault.config.JdkTls.INSECURE_TRUST_MANAGER;
+import static org.assertj.core.api.Assertions.assertThat;
+
+class InsecureTlsTest {
+
+    @Test
+    void testInsecureTlsDisabled() throws NoSuchAlgorithmException {
+        JdkTls insecureTls = new JdkTls(new Tls(null, new InsecureTls(false)));
+        assertThat(insecureTls.sslContext()).isSameAs(SSLContext.getDefault());
+    }
+
+    @Test
+    void testInsecureTlsEnabled() {
+        InsecureTls insecureTls = new InsecureTls(true);
+        Optional<TrustManager[]> maybeTrustManagers = JdkTls.getTrustManagers(insecureTls);
+        assertThat(maybeTrustManagers).isNotEmpty().hasValueSatisfying(trustManagers -> {
+            assertThat(trustManagers).hasSize(1).containsExactly(INSECURE_TRUST_MANAGER);
```

Comment:
non-blocking.. the fact that it is using *the* INSECURE_TRUST_MANAGER is really an implementational concern.  The unit test shouldn't care.  A better way would be to create a certificate and offer it to the trust managers and make sure they accept it.

Reply:
Have updated the tests to check that the TrustManagers reject or accept a self-signed-cert